### ISSUE TITLE
Invocation of ~/.config/direnv/direnvrc is not recursive

### ISSUE THREAD

**Issue by garthkerr** [direnv/direnv#606]
**Is your feature request related to a problem? Please describe.**
I debated filing this as a bug or a feature request. I believe the current _single_ invocation of `direnvrc` in the current working directory is the intended behavior (unless I'm misunderstanding the behavior). That said, I tried doing an implicit `source_up` within a specific path:

```
[[ "$PWD" == "${HOME}/Workspaces/"* ]] && source_up
```

**Describe the solution you'd like**
It would be nice if there was a way to differentiate `~/.config/direnv/direnvrc` invocations in the current `$PWD` and at every `.envrc` encountered with a _stacking_ `source_up` (perhaps with a different config file). This would also be useful with this proposal https://github.com/direnv/direnv/issues/556 where additional private/local files could be stacked recursively.

**Comment by zimbatm** [comment]
yeah that's an interesting idea. this would require to load each `.envrc` in isolation which is not currently the case.

I don't think I will implement that at the moment but it's good to keep in mind if I ever write a direnv 3.0
